### PR TITLE
fix: tecdsa

### PR DESCRIPTION
**Pull request by benjizhai**
- Fix tecdsa bugs
- Fix stable memory bound
- Fix chain id type
- Update did file

### CODE REVIEW

File: rs/minter/minter.did [hunk 1]
```diff
@@ -55,6 +63,19 @@ type ReturnError = variant {
...
+  get_version_text : () -> (text) query;
+  grant_admin : (principal) -> ();
+  is_admin : (principal) -> (bool) query;
+  is_owner : (principal) -> (bool) query;
+  owner : () -> (opt principal) query;
+  pending_owner : () -> (opt principal) query;
+  renounce_admin : () -> ();
+  revoke_admin : (principal) -> ();
+  transfer_ownership : (opt principal) -> ();
+  transfer_ownership_immediate : (opt principal) -> ();
```
Comment (ninegua) [inline rs/minter/minter.did:78]: How did you generate these declarations? The Makefile uses candid-extractor, but it doesn't seem to produce these functions.

Comment (benjizhai) [reply rs/minter/minter.did:78]: Those endpoints are exposed by the Rustic library. https://github.com/internet-computer-protocol/rustic#export-candid

Comment (benjizhai) [reply rs/minter/minter.did:78]: For example, `transfer_ownership` is defined here https://github.com/internet-computer-protocol/rustic/blob/653c4057687143db4d8d527f5e127e1fcd6e2b1e/src/access_control.rs#L124

Comment (ninegua) [reply rs/minter/minter.did:78]: Interesting.. so it has to be combined manually? Hope this can be improved in the future ...

**Review by ninegua — APPROVED** [review]

**Pull request merged**